src/main/java/com/es/phoneshop/DAO/CustomProductDao.java

```diff
@@ -14,31 +15,27 @@
 import java.util.stream.Collectors;
 
 public class CustomProductDao implements ProductDao {
-    private static final ReadWriteLock lock = new ReentrantReadWriteLock();//is it a good variant to make locks static?
+    private static final ReadWriteLock lock = new ReentrantReadWriteLock();
```

Comment:
you can make it non static and for the singleton initialization use the synchronized, in this case we can use synchronized in good for performance way